### PR TITLE
Remove platform.linux_distribution()

### PR DESCRIPTION
**Pull request by tivek**
Related to #912. Use the `distro` package instead of deprecated `platform.linux_distribution()`.

Before merging it would probably be worthwhile to check if `OSInfo.os_version` gives the same value for a couple of older distros.

Also, can we replace `OSInfo.get_debian_version_name()` with `distro.codename()`? Careful: `distro` needs Python 2.6 which means it will not work pre-Debian 6.0 (Squeeze). Is Conan even used on such old distributions at all?

### CODE REVIEW

**Comment by lasote** [comment]
Yes, we have to check some things before merging, could be little dangerous.
Just a note: linux_distribution() is deprecated since version 3.5, will be removed in version 3.7.
Thanks!!!!

**Comment by tivek** [comment]
> Yes, we have to check some things before merging, could be little dangerous.
>Just a note: linux_distribution() is deprecated since version 3.5, will be removed in version 3.7.
Yes. I thought an early heads-up would be useful to work out all the kinks.

>About your comment related to old distributions, you never know which distributions are using your users but I'm sure that someone is using conan with Debian < Squeeze
True, true, no need to be overeager :)

Let me know if I can help with testing. Any plans to try out some other Linux environments in TravisCI? Long-term It might be useful to have a couple other LTS distros in the matrix.

File: conans/tools.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import requests
 from conans.client.output import ConanOutput
 import platform
+if platform.system() == "Linux": import distro
```
Comment (lasote) [inline conans/tools.py]: Remove this line and move the import to line 255 (inside if self.is_linux block)

**Comment by lasote** [comment]
I would prefer to keep the "distro" in the regular requirements.txt file. Another file affects to package creators, deploys, etc. You can safe install "distro" package in Windows and OSX, you just have to protect the import (as you did). What do you think @memsharded ?

File: .ci/travis/install.sh [hunk 2]
```diff
@@ -40,6 +40,8 @@ if [[ "$(uname -s)" == 'Darwin' ]]; then
     esac
     pyenv rehash
     pyenv activate conan
+else
```
Comment (lasote) [inline .ci/travis/install.sh]: not needed anymore

File: setup.py [hunk 3]
```diff
@@ -28,6 +28,8 @@ def get_requires(filename):
 project_requirements = get_requires("conans/requirements.txt")
 if platform.system() == "Darwin":
     project_requirements.extend(get_requires("conans/requirements_osx.txt"))
+elif platform.system() == "Linux":
```
Comment (lasote) [inline setup.py]: Not needed anymore

File: .ci/travis/install.sh [hunk 4]
```diff
@@ -40,6 +40,8 @@ if [[ "$(uname -s)" == 'Darwin' ]]; then
...
+    pip install -r conans/requirements_linux.txt
```
Comment (memsharded) [inline .ci/travis/install.sh]: So finally this requirements_linux is not necessary if moving "distro" to the main requirements.txt file?

File: setup.py [hunk 5]
```diff
@@ -28,6 +28,8 @@ def get_requires(filename):
...
+    project_requirements.extend(get_requires("conans/requirements_linux.txt"))
```
Comment (memsharded) [inline setup.py]: same here

**Comment by tivek** [comment]
Sorry for the flurry of commits. When the code is in shape let me know to squash the branch before merging.

**Comment by memsharded** [comment]
No worries @tivek , everything looks good now, waiting for CI, and github has this "squash & merge button" that will to the squash for you. Thanks very much for your contribution!

**Pull request merged**